src/bindings/python/src/pyopenvino/core/infer_request.cpp

```diff
@@ -86,31 +103,91 @@ void regclass_InferRequest(py::module m) {
             self._request.start_async();
         },
         py::arg("inputs"),
-        py::arg("userdata"));
+        py::arg("userdata"),
+        R"(
+            Starts inference of specified input(s) in asynchronous mode.
+            Returns immediately. Inference starts also immediately.
+            Calling any method while the request is running will lead to
+            throwning exceptions.
+
+            Parameters
+            ----------
+            inputs : dict[Union[int, str, openvino.runtime.ConstOutput] : openvino.runtime.Tensor]
+                Data to set on input tensors.
+
+            userdata : Any
+                Any data that will be passed inside callback call.                
+
+            Returns
+            ----------
+            start_async : None
+        )");
+
+    cls.def(
+        "cancel",
+        [](InferRequestWrapper& self) {
+            self._request.cancel();
+        },
+        R"(
+            Cancels inference request.
 
-    cls.def("cancel", [](InferRequestWrapper& self) {
-        self._request.cancel();
-    });
+            Parameters
+            ----------
+            None
 
-    cls.def("wait", [](InferRequestWrapper& self) {
-        py::gil_scoped_release release;
-        self._request.wait();
-    });
+            Returns
+            ----------
+            cancel : None
+        )");
+
+    cls.def(
+        "wait",
+        [](InferRequestWrapper& self) {
+            py::gil_scoped_release release;
+            self._request.wait();
+        },
+        R"(
+            Waits for the result to become available. 
+            Blocks until the result becomes available.
+
```

Comment:
I think we also should note that this function releases the GIL, something like that - `This function releases the GIL, so another
  python thread can work while running 
  this function in the background.` we can add in all places when we release it.